xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/ImageDecoder.h

```diff
@@ -0,0 +1,163 @@
+    ///                                 id ADDON_INSTANCE_IMAGEDECODER.
+    CInstanceImageDecoder(KODI_HANDLE instance)
+      : IAddonInstance(ADDON_INSTANCE_IMAGEDECODER)
+    {
+      if (CAddonBase::m_interface->globalSingleInstance != nullptr)
+        throw std::logic_error("kodi::addon::CInstanceImageDecoder: Creation of multiple together with single instance way is not allowed!");
+
+      SetAddonStruct(instance);
+    }
+    //--------------------------------------------------------------------------
+
+    virtual ~CInstanceImageDecoder() = default;
+
+    //==========================================================================
+    /// \brief Initialize an encoder
+    ///
+    /// \param[in] buffer The data to read from memory
+    /// \param[in] bufSize The buffer size
+    /// \param[in,out] width The optimal width of image on entry, obtained width on return
+    /// \param[in,out] height The optimal height of image, actual obtained height on return
+    /// \return true if successful done, false on error
+    ///
+    virtual bool LoadImageFromMemory(unsigned char* buffer, unsigned int bufSize,
+                                     unsigned int& width, unsigned int& height) = 0;
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    /// \brief Decode previously loaded image
+    ///
+    /// \param[in] pixels Output buffer
+    /// \param[in] width Width of output image
+    /// \param[in] height Height of output image
+    /// \param[in] pitch Pitch of output image
+    /// \param[in] format Format of output image
+    /// \return true if successful done, false on error
+    ///
+    virtual bool Decode(unsigned char* pixels,
+                        unsigned int width, unsigned int height,
+                        unsigned int pitch, ImageFormat format) = 0;
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    /// \brief Get the wanted mime type from Kodi
+    ///
+    /// \return the mimetype wanted from Kodi
+    ///
+    inline std::string MimeType() { return m_instanceData->props.mimetype; }
+    //--------------------------------------------------------------------------
+
+  private:
+    void SetAddonStruct(KODI_HANDLE instance)
+    {
+      if (instance == nullptr)
+        throw std::logic_error("kodi::addon::CInstanceImageDecoder: Creation with empty addon structure not allowed, table must be given from Kodi!");
+
+      m_instanceData = static_cast<AddonInstance_ImageDecoder*>(instance);
+      m_instanceData->toAddon.addonInstance = this;
+      m_instanceData->toAddon.load_image_from_memory = ADDON_LoadImageFromMemory;
+      m_instanceData->toAddon.decode = ADDON_Decode;
+    }
+
+    inline static bool ADDON_LoadImageFromMemory(AddonInstance_ImageDecoder const* instance,
```

Comment:
"type const *" is the same as "const type *". Why do you use other syntax here than you do usually? Is your intention to denote that the pointer value itself is const and the data behind the pointer can be written? Then you must use "type * const". Applies also to other methods in the PR.